lib/models.py

```diff
@@ -2,13 +2,231 @@
 import openai
 from typing import Any, Callable, List, Optional
 
+import numpy as np
+import itertools
+
 class CoT():
```

Comment:
Change name to ToT (tree of thought, and change relevant comments in the class here).

Add an inherited class CoT

```
class CoT(ToT):
  def __init__(self):
      super().__init__()
      self.num_steps = 1 if num_steps is None else num_steps
      self.num_generate = …. # whatever variables need to be overridden to guarantee chain of thought
```

Reply:
Print when overriding a kwarg